src/hooks/useOrders.ts

```diff
@@ -27,19 +28,130 @@ function filterDeletedOrders(orders: AuctionElement[]): AuctionElement[] {
   )
 }
 
-export function useOrders(): AuctionElement[] {
+interface Result {
+  orders: AuctionElement[]
+  forceOrdersRefresh: () => void
+  isLoading: boolean
+}
+
+export function useOrders(): Result {
   const { userAddress, networkId, blockNumber } = useWalletConnection()
   const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+  const [offset, setOffset] = useSafeState<number | undefined>(0)
+  const [isLoading, setIsLoading] = useSafeState<boolean>(false)
+
+  // Tracking last seen values to decide whether to update
+  const [lastBlockNumber, setLastBlockNumber] = useSafeState<number | undefined>(undefined)
+  const [lastNetworkId, setLastNetworkId] = useSafeState<number | undefined>(undefined)
+  const [lastUserAddress, setLastUserAddress] = useSafeState<string | undefined>(undefined)
+
+  const _fetchOrders = useCallback(
+    async (
+      userAddress: string | undefined,
+      networkId: number | undefined,
+      offset: number | undefined,
+      areThereOrders?: boolean,
+    ): Promise<void> => {
+      if (userAddress === undefined || networkId === undefined || offset === undefined) {
+        return
+      }
+
+      await exchangeApi.getOrdersPaginated({ userAddress, networkId, offset }).then(({ orders, nextIndex }) => {
+        if (orders.length) {
```

Comment:
this is `orders.length !== 0` right? or there's a case where orders.length is `undefined`?

Reply:
It should always return an array, even when empty: https://github.com/gnosis/dex-react/blob/develop/src/api/exchange/ExchangeApi.ts#L140

Reply:
so maybe would be slightly better to do `orders.length > 0` or `orders.length !== 0`?

Reply:
Alright